views.go

```diff
@@ -0,0 +1,221 @@
+package slack
+
+import (
+	"context"
+	"encoding/json"
+)
+
+const (
+	VTModal   ViewType = "modal"
+	VTHomeTab ViewType = "home"
+)
+
+type ViewType string
+
+type View struct {
+	SlackResponse
+	ID              string           `json:"id"`
+	TeamID          string           `json:"team_id"`
+	Type            ViewType         `json:"type"`
+	Title           *TextBlockObject `json:"title"`
+	Close           *TextBlockObject `json:"close"`
+	Submit          *TextBlockObject `json:"submit"`
+	Blocks          Blocks           `json:"blocks"`
+	PrivateMetadata string           `json:"private_metadata"`
+	CallbackID      string           `json:"callback_id"`
+	State           interface{}      `json:"state"`
```

Comment:
The type of `state` field is not clear so far.

Reply:
The `state` is used in `view_submission` interaction payloads. It's a dictionary of dictionaries, the top level dictionary contains BlockID keys and the nested dictionary contains ActionID keys which map the type and value of the user input on a block element:
https://api.slack.com/reference/interaction-payloads/views#view_submission

I'm using the following in my fork to represent that:
```
type ElementState struct {
	Type  string `json:"type"`
	Value string `json:"value"`
}

type ViewState struct {
	Values map[string]map[string]ElementState `json:"values"`
}
```

Which supports the following JSON structure:
```
"state": {
    "values": {
        "multi-line": {
            "ml-value": {
                "type": "plain_text_input",
                "value": "This is my example inputted value"
            }
        }
    }
}
```

Reply:
Good catch! Thanks!
I think this usage of `state` is recently update and Slack updates its API day by day.
So I want to merge this PR as soon as possible because this PR becomes larger as time passes 😢

Reply:
Yeah it's very much a moving target 😂

In fact my above snippet was wrong, or it only works for text input. Here's what I should have used instead:

```
type ViewState struct {
	Values map[string]map[string]BlockAction `json:"values"`
}
```

`BlockAction` already existed and handles unmarshalling for all block actions (including selects, text, date pickers, etc.)